### PR TITLE
[DUOS-342][risk=low] Update user role table for DAC requirements

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadinstitute.atlassian.net/browse/DUOS-342

## Changes
The primary goal of this PR is to refactor the user role table to allow for users to be in multiple Data Access Committees. Users should be able to have roles in different DACs as well as having the roles of Researcher, Data Owner, etc. There are no functional changes to any existing queries - this is primarily a refactoring PR, hence the large number of small changes. 

* New PK: `user_role_id`
* `roleId` -> `role_id`
* `dacUserId` -> `user_id` ... have not touched the `dac_user` table itself, that's a follow on ticket.
* FK for `dac_id` to the newer data access committee table in https://github.com/DataBiosphere/consent/pull/346

Secondarily, we're updating tests so that we can separate unit testing against Resources vs DAOs. I am hopeful that this strategy will allow for easier testing in future PRs. For the new DAO tests, we have a test for every dao call in the classes that were touched. These are all new tests, many will need to be removed in upcoming stories, but for now, they are here to ensure that there are no broken queries from the table refactor.

Follow-on stories for bugs found while working on this:
* https://broadinstitute.atlassian.net/browse/DUOS-392
* https://broadinstitute.atlassian.net/browse/DUOS-393

## Testing
Since this is a risky table migration task, I have tested this in the following ways:
* Comprehensive query unit tests.
* Exported a clone of dev, ran the liquibase changes against that, successfully.
* Exported a clone of production, ran the liquibase changes against that, successfully.
* Created a clone in GCP, ran the liquibase changes against that, successfully.
* Stood up a local API instance againsted updated clone and manually tested.
* Stood up a local UI instance against local API instance and manually tested.

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/db/UserRoleMapper.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public class UserRoleMapper implements ResultSetMapper<UserRole> {
+    public UserRole map(int index, ResultSet r, StatementContext ctx) throws SQLException {
+        return new UserRole(
+                r.getInt("role_id"),
+                r.getString("name"),
+                r.getBoolean("email_preference"),
+                r.getString("rationale"),
+                RoleStatus.getStatusByValue(r.getInt("status")));
+    }
+}
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/UserRoleMapper.java:19]: This is really just a move from `RoleMapper` to `UserRoleMapper`.

File: src/main/java/org/broadinstitute/consent/http/service/DatabaseDataSetAPI.java [hunk 2]
```diff
@@ -461,11 +461,8 @@ private void processAssociation(List<DataSet> dataSetList) {
     }
 
 
-    private boolean userIs(String rol, Integer dacUserId) {
-        if (dsRoleDAO.findRoleByNameAndUser(rol, dacUserId) != null) {
-            return true;
-        }
-        return false;
+    private boolean userIs(String roleName, Integer dacUserId) {
+        return userRoleDAO.findRoleByNameAndUser(roleName, dacUserId) != null;
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DatabaseDataSetAPI.java:465]: minor optimization here ... couldn't resist.

File: src/main/java/org/broadinstitute/consent/http/db/DACUserDAO.java [hunk 3]
```diff
@@ -15,31 +15,33 @@
...
+    @SqlQuery("select u.* from dacuser u inner join user_role du on du.user_id = u.dacUserId inner join roles r on r.roleId = du.role_id where r.name = :roleName")
     List<DACUser> describeUsersByRole(@Bind("roleName") String roleName);
 
-    @SqlQuery("select u.dacUserId from dacuser u inner join user_role du on du.dacUserId = u.dacUserId inner join roles r on r.roleId = du.roleId where u.dacUserId = :dacUserId and r.name = 'Chairperson'")
+    @SqlQuery("select u.dacUserId from dacuser u inner join user_role du on du.user_id = u.dacUserId inner join roles r on r.roleId = du.role_id where u.dacUserId = :dacUserId and r.name = 'Chairperson'")
     Integer checkChairpersonUser(@Bind("dacUserId") Integer dacUserId);
 
     @Mapper(DACUserRoleMapper.class)
-    @SqlQuery("select u.*,r.roleId, r.name from dacuser u inner join user_role du on du.dacUserId = u.dacUserId inner join roles r on r.roleId = du.roleId where r.name = 'Chairperson' or r.name = 'Member'")
+    @SqlQuery("select u.*, r.roleId, r.name from dacuser u inner join user_role du on du.user_id = u.dacUserId inner join roles r on r.roleId = du.role_id where r.name = 'Chairperson' or r.name = 'Member'")
```
Comment (dags) [inline src/main/java/org/broadinstitute/consent/http/db/DACUserDAO.java:40]: r.roleId should be r.role_id ?

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/db/DACUserDAO.java:40]: `r.roleId` refers to the `roles` table which is unchanged in this PR. It should change relatively soon, but not at this point.

**Review by dags — APPROVED** [review]
All looks good!

**Pull request merged**